csv/result.go

```diff
@@ -327,30 +327,44 @@ func readMetadata(r *csv.Reader, c ResultDecoderConfig, extraLine []string) (tab
 		if n != len(line) {
 			return tableMetadata{}, errors.Wrap(csv.ErrFieldCount, codes.Invalid, "failed to read annotations")
 		}
-		switch annotation := strings.TrimPrefix(line[annotationIdx], commentPrefix); annotation {
-		case datatypeAnnotation:
-			datatypes = copyLine(line[recordStartIdx:])
-		case groupAnnotation:
-			groups = copyLine(line[recordStartIdx:])
-		case defaultAnnotation:
-			resultID = line[resultIdx]
-			tableID = line[tableIdx]
-			if _, err := strconv.ParseInt(tableID, 10, 64); tableID != "" && err != nil {
-				return tableMetadata{}, fmt.Errorf("default Table ID is not an integer")
-			}
-			defaults = copyLine(line[recordStartIdx:])
-		default:
-			if annotation == "" {
-				switch {
-				case datatypes == nil:
-					return tableMetadata{}, fmt.Errorf("missing expected annotation datatype")
-				case groups == nil:
-					return tableMetadata{}, fmt.Errorf("missing expected annotation group")
-				case defaults == nil:
-					return tableMetadata{}, fmt.Errorf("missing expected annotation default")
+		if strings.HasPrefix(line[annotationIdx], commentPrefix) {
+			switch annotation := strings.TrimPrefix(line[annotationIdx], commentPrefix); annotation {
+			case datatypeAnnotation:
+				datatypes = copyLine(line[recordStartIdx:])
+			case groupAnnotation:
+				groups = copyLine(line[recordStartIdx:])
+			case defaultAnnotation:
+				resultID = line[resultIdx]
+				tableID = line[tableIdx]
+				if _, err := strconv.ParseInt(tableID, 10, 64); tableID != "" && err != nil {
+					return tableMetadata{}, fmt.Errorf("default Table ID is not an integer")
 				}
+				defaults = copyLine(line[recordStartIdx:])
+			default:
+				if annotation == "" {
+					switch {
+					case datatypes == nil:
+						return tableMetadata{}, fmt.Errorf("missing expected annotation datatype")
+					case groups == nil:
+						return tableMetadata{}, fmt.Errorf("missing expected annotation group")
+					case defaults == nil:
+						return tableMetadata{}, fmt.Errorf("missing expected annotation default")
+					}
+				}
+				// Skip extra annotation
+			}
+		} else {
```

Comment:
Is there a way you can refactor this so the above section that returns errors for missing annotations uses this code instead? I'd also like it if we could reduce the indentation since the indentation is getting to a bit too deep for my personal taste.

Reply:
Yeah, I wasn't sure how much urgency there was here. I'm happy to re-phrase that entire chunk of code, as it's already not very clear exactly what's going on.